src/screens/Rari/RariWithdraw.js

```diff
@@ -0,0 +1,285 @@
+`;
+
+const RariWithdrawScreen = ({
+  assets, navigation, feeInfo, isEstimating, estimateErrorMessage,
+  resetEstimateTransaction, balances, setEstimatingTransaction, calculateRariWithdrawTransactionEstimate,
+  activeAccountAddress, supportedAssets,
+}) => {
+  useEffect(() => {
+    resetEstimateTransaction();
+  }, []);
+
+  const [selectedAsset, setSelectedAsset] = useState(assets[ETH]);
+  const [assetValue, setAssetValue] = useState('');
+  const [inputValid, setInputValid] = useState(false);
+  const [transactionData, setTransactionData] = useState(null);
+  const [isCalculatingMaxAmount, setIsCalculatingMaxAmount] = useState(false);
+  const [customBalances, setCustomBalances] = useState({});
+
+
+  const rariPool = navigation.getParam('rariPool');
+
+  let notEnoughForFee = false;
+  if (feeInfo && parseFloat(assetValue)) {
+    notEnoughForFee = !isEnoughBalanceForTransactionFee(balances, {
+      txFeeInWei: feeInfo.fee,
+      amount: assetValue,
+      decimals: selectedAsset?.decimals,
+      symbol: selectedAsset?.symbol,
+      gasToken: feeInfo.gasToken,
+    });
+  }
+
+  const errorMessage = notEnoughForFee
+    ? t('error.notEnoughTokenForFee', { token: feeInfo?.gasToken?.symbol || ETH })
+    : estimateErrorMessage;
+
+  useEffect(() => {
+    if (!assetValue || !parseFloat(assetValue) || !selectedAsset) return;
+    setEstimatingTransaction(true);
+    getRariWithdrawTransaction(rariPool, activeAccountAddress, assetValue, selectedAsset)
+      .then(txsAndExchangeFee => {
+        if (!txsAndExchangeFee) {
+          Toast.show({
+            message: t('toast.rariServiceFailed'),
+            emoji: 'hushed',
+          });
+          return;
+        }
+        const { withdrawTransaction, exchangeFeeBN: _exchangeFeeBN, slippage: _slippage } = txsAndExchangeFee;
+        setTransactionData({
+          transactionPayload: withdrawTransaction,
+          exchangeFeeBN: _exchangeFeeBN,
+          slippage: _slippage,
+        });
+        calculateRariWithdrawTransactionEstimate(withdrawTransaction);
+      })
+      .catch((error) => {
+        reportErrorLog('Rari service failed: Error creating transaction payload', { error });
+        Toast.show({
+          message: t('toast.rariServiceFailed'),
+          emoji: 'hushed',
+        });
+      });
+  }, [assetValue, selectedAsset]);
+
+  useEffect(() => {
+    if (!selectedAsset) return;
+    setIsCalculatingMaxAmount(true);
+    // we set balance to a very high number to avoid showing user "insufficient balance" error
+    // when switching assets before we even fetch the max
+    setCustomBalances({ [selectedAsset.symbol]: { symbol: selectedAsset.symbol, balance: 1e18 }, ...customBalances });
+    getMaxWithdrawAmount(rariPool, selectedAsset, activeAccountAddress)
+      .then((maxWithdrawAmount) => {
+        if (!maxWithdrawAmount) {
+          Toast.show({
+            message: t('toast.rariServiceFailed'),
+            emoji: 'hushed',
+          });
```

Comment:
One branch without state reset left:
```suggestion
          });
          setIsCalculatingMaxAmount(false);
```

(alternatively, there could be another `.then` at the end, in place of `Promise.finally`)